types/agent.go

```diff
@@ -54,15 +55,16 @@ const (
 
 // RegisterAgent request payload for registering a new agent.
 type RegisterAgent struct {
-	Name      string           `json:"name"`
-	MachineID string           `json:"machineID"`
-	Type      AgentType        `json:"type"`
-	Version   string           `json:"version"`
-	Edition   AgentEdition     `json:"edition"`
-	Flags     []string         `json:"flags"`
-	RawConfig string           `json:"rawConfig"`
-	Metadata  *json.RawMessage `json:"metadata"`
-	Tags      []string         `json:"tags"`
+	Name          string           `json:"name"`
+	MachineID     string           `json:"machineID"`
+	Type          AgentType        `json:"type"`
+	Version       string           `json:"version"`
+	Edition       AgentEdition     `json:"edition"`
+	Flags         []string         `json:"flags"`
+	RawConfig     string           `json:"rawConfig"`
+	Metadata      *json.RawMessage `json:"metadata"`
+	Tags          []string         `json:"tags"`
+	EnvironmentID string           `json:"environmentID"`
```

Comment:
This is taken from the token.

Reply:
Not really, an environment id has to be provided when registering an agent, that's an explicit action. We don't need environment specific actions, if the environmentID isn't provided, then we default to "default" env. We will later extend the calyptia plugin to allow passing a custom environment name.

In fact, this parameter should be a Environment string rather than an ID.